packages/grafana-data/src/types/action.ts

```diff
@@ -0,0 +1,71 @@
+import { ScopedVars } from './ScopedVars';
+import { DataFrame, Field, ValueLinkConfig } from './dataFrame';
+import { InterpolateFunction } from './panel';
+import { SelectableValue } from './select';
+
+export interface Action<T = ActionType.Fetch, TOptions = FetchOptions> {
+  type: T;
+  title: string;
+  options: TOptions;
+}
+
+/**
+ * Processed Action Model. The values are ready to use
+ */
+export interface ActionModel<T = any> {
+  title: string;
+  onClick: (event: any, origin?: any) => void;
+}
+
+interface FetchOptions {
+  method: HttpRequestMethod;
+  url: string;
+  body?: string;
+  queryParams?: Array<[string, string]>;
+  headers?: Array<[string, string]>;
+}
+
+export enum ActionType {
+  Fetch = 'fetch',
```

Comment:
maybe it's better to have just a constant instead of an enum in this case?

Reply:
added it as an enum as the list is going to grow soon 😁